test/test_dynamic_transform.cpp

```diff
@@ -0,0 +1,485 @@
+
+  auto reshape_shape0 = IrBuilder::create<Int>();
+  fusion.addInput(reshape_shape0);
+  auto reshape_shape1 = IrBuilder::create<Int>();
+  fusion.addInput(reshape_shape1);
+
+  auto tv2 = reshape(tv0, {reshape_shape0, reshape_shape1});
+  auto tv3 = add(tv1, tv2);
+
+  fusion.addOutput(tv3);
+
+  fusion.printMath();
+
+  ExpressionEvaluator expr_eval;
+
+  // input: 4, 3
+  // output: 3, 4
+  expr_eval.bind(tv0->axis(0)->extent(), 4);
+  expr_eval.bind(tv0->axis(1)->extent(), 3);
+  expr_eval.bind(reshape_shape0, 3);
+  expr_eval.bind(reshape_shape1, 4);
+
+  auto info = DynamicTransform::getConcretizationInfo(&fusion, &expr_eval);
+
+  std::cerr << info.toString() << std::endl;
+
+  DynamicTransform::concretizeFusion(&fusion, info);
+
+  TORCH_CHECK(
+      !fusion.hasDynamicTransform(), "Expected to have no dynamic transform");
+
+  fusion.printMath();
+}
+
+TEST_F(NVFuserTest, DynamicTransform4_CUDA) {
+#if 1
+  std::vector<std::pair<std::vector<int64_t>, std::vector<int64_t>>>
+      before_after_shapes = {
+          {{4, 3}, {3, 4}},
+          {{4, 3}, {12, 1}},
+      };
+#else
+  std::vector<std::pair<std::vector<int64_t>, std::vector<int64_t>>>
+      before_after_shapes = {
+          {{4, 3}, {12, 1}},
+      };
+#endif
+  for (auto before_after : before_after_shapes) {
+    std::cerr << "Before: " << before_after.first << std::endl;
+    std::cerr << "After: " << before_after.second << std::endl;
+    Fusion fusion;
+    FusionGuard fg(&fusion);
+
+    auto tv0 = makeSymbolicTensor(2);
+    fusion.addInput(tv0);
+    auto tv1 = makeSymbolicTensor(2);
+    fusion.addInput(tv1);
+
+    auto reshape_shape0 = IrBuilder::create<Int>();
+    fusion.addInput(reshape_shape0);
+    auto reshape_shape1 = IrBuilder::create<Int>();
+    fusion.addInput(reshape_shape1);
+
+    auto tv2 = reshape(tv0, {reshape_shape0, reshape_shape1});
+    // tv3 will also have symbolic axes
+    auto tv3 = set(tv2);
+    auto tv4 = add(tv1, tv3);
+
+    fusion.addOutput(tv4);
+
+    fusion.printMath();
+
+    ExpressionEvaluator expr_eval;
+
+    expr_eval.bind(tv0->axis(0)->extent(), before_after.first.at(0));
+    expr_eval.bind(tv0->axis(1)->extent(), before_after.first.at(1));
+    expr_eval.bind(tv2->axis(0)->extent(), before_after.second.at(0));
+    expr_eval.bind(tv2->axis(1)->extent(), before_after.second.at(1));
+
+    auto info = DynamicTransform::getConcretizationInfo(&fusion, &expr_eval);
+
+    std::cerr << info.toString() << std::endl;
+
+    DynamicTransform::concretizeFusion(&fusion, info);
+
+    TORCH_CHECK(
+        !fusion.hasDynamicTransform(), "Expected to have no dynamic transform");
+
+    fusion.printMath();
+  }
+}
+
+// Dynamic reshape followed by static resize
+TEST_F(NVFuserTest, DynamicTransform5_CUDA) {
+  std::vector<std::pair<std::vector<int64_t>, std::vector<int64_t>>>
+      before_after_shapes = {
+          {{4, 3}, {3, 4}},
+          //{{4, 3}, {12, 1}}, not possible to do pad a broadcast domain yet
```

Comment:
I thought pad on a broadcast domain has been patched?!

Reply:
Not yet